### ISSUE TITLE
Workflow redirect and notifications

### ISSUE THREAD

**Issue by Hazzamanic** [OrchardCMS/Orchard#5841]
When creating a workflow that ends in a redirect, all notifications added in the workflow prior to the redirect are lost. Is this the correct behaviour? It makes sense that it is because it creates a new context for the redirect but I've seen quite a few peoples example workflows (e.g. dynamic forms on ideliverable: http://www.ideliverable.com/blog/customizing-user-registration-and-login-with-dynamic-forms-and-workflows) using notifications and redirects together.

I'm running 1.9.1

**Comment by Piedone** [comment]
No, notifications (added through INotifier) shouldn't be gone after a redirect. Are you sure there is only a single redirect (as notifications are only displayed on the next page load, but not after that)? Are any other notifications you get e.g. on the admin working properly?

**Comment by Hazzamanic** [comment]
Yeah, the notifications are added, the redirect is made, but it still completes the original request, so notifications will be fired there (did that make sense?).

The key is to prevent the controller from returning a result:

if (Response.IsRequestBeingRedirected) return new EmptyResult();

I don't have access to the rest of Orchard's source but I assume controllers where workflows expecting a redirect in them contain this (or similar) line of code?
